tst/functional-tests/framework/fake-servers/control_plane/signaling_server.py

```diff
@@ -0,0 +1,192 @@
+from flask import Flask, request, json, make_response
+from channel import channel
+
+app = Flask(__name__)
+
+channelMap = {}
+channelArnMap = {}
+
+@app.route("/abc", methods = ['GET'])
+def hello_world():
+    return "<p>Hello, World!</p>"
+
+@app.route("/test_post", methods = ['POST'])
+def hellow_world_post():
+    data = json.loads(request.data)
+    user = data['name']
+    return f'<p>Hello, World! {user} </p>'
+
+## ===================================================================== ##
+# ===================== Describe Signaling Channel ====================== #
+## ===================================================================== ##
+
+def handleDescribeSignalingChannelRequest(channelName, channelArn):
+    if channelName not in channelMap:
+        responseMessage = {}
+        responseMessage['Message'] = 'The requested channel is not found or not active.'
+        return app.response_class(
+            response = json.dumps(responseMessage),
+            status = 404,
+            mimetype='application/json'
+        )
+
+    ch = channelMap[channelName]
+    if (channelArn != '' and channelArn != ch.channelArn):
+        responseMessage = {}
+        responseMessage['Message'] = f'Channel name/arn mismatch name = {channelName}, expected arn = {channelArn}, actual arn = {ch.channelArn}'
+        return app.response_class(
+            response = json.dumps(responseMessage),
+            status = 400,
+            mimetype='application/json'
+        )
+
+    content = {}
+    content['ChannelInfo'] = ch.getChannelInfo()
+    return app.response_class(
+        response = json.dumps(content),
+        status = 200,
+        mimetype='application/json'
+    )
+
+@app.route("/describeSignalingChannel", methods = ['POST'])
+def describeSignalingChannelHandler():
+    requestPayload = request.json
+
+    channelArn = ''
+    if 'ChannelARN' in requestPayload:
+        channelArn = requestPayload['ChannelARN']
+
+    channelName = ''
+    if 'ChannelName' in requestPayload:
+        channelName = requestPayload['ChannelName']
+
+    if channelName == '' and channelArn == '':
```

Comment:
I think this should be an OR condition? This would mean if `channelName` exists and `channelArn` does not or vice versa, we would not be returning a `400`, which does not seem right.